entry_types/scrolled/package/spec/support/rendering.js

```diff
@@ -27,11 +43,41 @@ import {normalizeSeed} from './normalizeSeed';
  *   is passed as a second parameter.
  */
 export function renderInEntry(ui, {seed, setup, wrapper, ...options} = {}) {
-  return render(ui,
-                {
-                  wrapper: createWrapper(seed, setup, wrapper),
-                  ...options
-                });
+  options = {
+    wrapper: createWrapper(seed, setup, wrapper),
+    ...options
+  }
+
+  if (typeof ui === 'function') {
+    // Evaluate `ui` inside a React component to allow using hooks in
+    // the test. We also could have used `ui` as a React component
+    // directly (e.g. `React.createElement(ui)`). But when calling
+    // `rerender` with a different function than the one passed to the
+    // original `renderInEntry` call, e.g.
+    //
+    //   const {rerender} = renderInEntry(() => <MyComponent someProp={1} />);
+    //   rerender(() => <MyComponent someProp={2} />);
+    //
+    // React would unmount the `MyComponent` component from the first
+    // render call and mount a new one. We therefore define a single
+    // component that is reused across rerenders to ensure
+    // `MyComponent` stays mounted and just receives new props.
```

Comment:
**praise:** This comment answers very well precisely the questions about 'why?' that came up when I read the code below